lib/core/presentation/pages/event/event_control_panel_page/sub_pages/event_cohosts_page/event_cohosts_page.dart

```diff
@@ -0,0 +1,167 @@
+import 'package:app/core/application/event/edit_event_detail_bloc/edit_event_detail_bloc.dart';
+import 'package:app/core/application/event/event_detail_cohosts_bloc/event_detail_cohosts_bloc.dart';
+import 'package:app/core/application/event/event_provider_bloc/event_provider_bloc.dart';
+import 'package:app/core/application/event/manage_event_cohost_requests_bloc/manage_event_cohost_requests_bloc.dart';
+import 'package:app/core/data/user/dtos/user_dtos.dart';
+import 'package:app/core/domain/event/entities/event.dart';
+import 'package:app/core/domain/event/entities/event_cohost_request.dart';
+import 'package:app/core/domain/user/entities/user.dart';
+import 'package:app/core/presentation/pages/event/event_control_panel_page/sub_pages/event_cohosts_page/widgets/event_cohost_item.dart';
+import 'package:app/core/presentation/widgets/common/appbar/lemon_appbar_widget.dart';
+import 'package:app/core/presentation/widgets/common/button/linear_gradient_button_widget.dart';
+import 'package:app/i18n/i18n.g.dart';
+import 'package:app/router/app_router.gr.dart';
+import 'package:app/theme/spacing.dart';
+import 'package:app/theme/typo.dart';
+import 'package:auto_route/auto_route.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_screenutil/flutter_screenutil.dart';
+import 'package:flutter_vibrate/flutter_vibrate.dart';
+
+@RoutePage()
+class EventCohostsPage extends StatelessWidget {
+  const EventCohostsPage({
+    super.key,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    final event = context.read<EventProviderBloc>().event;
+    return EventCohostsPageView(event: event);
+  }
+}
+
+class EventCohostsPageView extends StatefulWidget {
+  const EventCohostsPageView({
+    super.key,
+    this.event,
+  });
+  final Event? event;
+
+  @override
+  State<EventCohostsPageView> createState() => _EventCohostsPageViewState();
+}
+
+class _EventCohostsPageViewState extends State<EventCohostsPageView> {
+  @override
+  Widget build(BuildContext context) {
+    final colorScheme = Theme.of(context).colorScheme;
+    final t = Translations.of(context);
+    return Scaffold(
+      appBar: LemonAppBar(
+        backgroundColor: colorScheme.onPrimaryContainer,
+        title: t.event.configuration.coHosts,
+      ),
+      backgroundColor: colorScheme.onPrimaryContainer,
+      resizeToAvoidBottomInset: true,
+      body: _buildContent(),
+    );
+  }
+
+  Widget _buildContent() {
+    final t = Translations.of(context);
+    List<EventCohostRequest> eventCohostsRequests =
+        context.watch<EventDetailCohostsBloc>().state.maybeWhen(
+              fetched: (eventCohostsRequests) => eventCohostsRequests,
+              orElse: () => [],
+            );
+    return BlocListener<ManageEventCohostRequestsBloc,
```

Comment:
can we move BlocListener in build method ? not recommend putting these logic inside utils method